### PR TITLE
Use IndexOf{Any} in a few more places

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Net.Mail/src/System/Net/Mail/MailBnfHelper.cs [hunk 1]
```diff
@@ -379,23 +379,12 @@ private static bool CheckForUnicode(char ch, bool allowUnicode)
...
-            {
-                if (data[i] == '\r' || data[i] == '\n')
-                {
-                    return true;
-                }
-            }
-            return false;
-        }
+        internal static bool HasCROrLF(string data) =>
+            data.AsSpan().IndexOfAny('\r', '\n') >= 0;
```
Comment (EgorBo) [inline src/libraries/System.Net.Mail/src/System/Net/Mail/MailBnfHelper.cs:387]: `IndexOfAny(CR, LF)` ?

File: src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs [hunk 2]
```diff
@@ -207,17 +207,8 @@ public override string ToString()
...
-            foreach (char character in value)
-            {
-                if ((character == HttpRuleParser.CR) || (character == HttpRuleParser.LF))
-                {
-                    return true;
-                }
-            }
-            return false;
-        }
+        private static bool ContainsNewLineCharacter(string value) =>
```
Comment (MihaZupan) [inline src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs:210]: We should be able to switch the one caller here to use [`HttpRuleParser.ContainsNewLine`](https://github.com/dotnet/runtime/blob/6a8f16e54fe2f91ef6457c94cab830f3c31baf9d/src/libraries/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs#L150) directly

File: src/libraries/System.Console/src/System/TermInfo.cs [hunk 3]
```diff
@@ -498,9 +498,8 @@ private static string ReadString(byte[] buffer, int pos)
             /// <summary>Finds the null-terminator for a string that begins at the specified position.</summary>
             private static int FindNullTerminator(byte[] buffer, int pos)
             {
-                int termPos = pos;
-                while (termPos < buffer.Length && buffer[termPos] != '\0') termPos++;
-                return termPos;
```
Comment (tannergooding) [inline src/libraries/System.Console/src/System/TermInfo.cs:503]: Just wondering, were these detected using any kind of analysis?

Is there potential for an analyzer here?

**Review by tannergooding — APPROVED** [review]

File: src/libraries/System.Console/src/System/TermInfo.cs (hunk 3 above)
Comment (stephentoub) [reply src/libraries/System.Console/src/System/TermInfo.cs:503]: These were just searching around for various patterns. I could have sworn we had an issue for considering an analyzer, though I can't find it now.

**Pull request merged**